### ISSUE TITLE
Restarting nomad after exec tasks complete results in cgroup errors

### ISSUE THREAD

**Issue by justenwalker** [hashicorp/nomad#176]
I'm trying to simulate some exec job completing and running it again. I'm using sleep of 30 seconds and using nomad run twice. It doesn't allocate any nodes the second time I run it, and if I restart the nomad client, it fails to bring itself back up.

## Steps to reproduce

This is on an Ubuntu 14.04. I have two machines - a server and client. I am running all of these commands on the client machine. My datacenter name is "us-east-1". 

Update: Steps 4 and 5 are not necessary to reproduce the error, just restarting nomad results in the error.

### 1. Run sleep-job

```
job "sleep-job" {
    datacenters = ["us-east-1"]
    # Rolling updates should be sequential
    update {
        stagger = "30s"
        max_parallel = 1
    }

    group "sleep-group" {
        count = 5

        # Sleep for a while
        task "sleep-task" {
            driver = "exec"
            config {
              command = "/bin/sleep"
              args = "30"
            }
	    resources {
              cpu = 50
	    }
        }
    }
}
```

```
$ nomad run sleep.nomad
==> Monitoring evaluation "18618691-4d6f-de21-b67c-f7d6d4929319"
    Evaluation triggered by job "sleep-job"
    Allocation "7e43b623-c82b-9f40-765c-0340a3d145a9" created: node "f4565321-4370-7a84-06ba-95ec80777c11", group "sleep-group"
    Allocation "833cc87a-bbe7-fc59-2ed2-6b3c72237621" created: node "f4565321-4370-7a84-06ba-95ec80777c11", group "sleep-group"
    Allocation "8ff5c0e5-a9e9-c846-5def-a813ff5eff92" created: node "f4565321-4370-7a84-06ba-95ec80777c11", group "sleep-group"
    Allocation "fbcdf7df-3d57-fc70-7353-cb1cb9a90540" created: node "f4565321-4370-7a84-06ba-95ec80777c11", group "sleep-group"
    Allocation "44640d26-7fc9-6f7d-3d21-2d063102f97d" created: node "f4565321-4370-7a84-06ba-95ec80777c11", group "sleep-group"
    Evaluation status changed: "pending" -> "complete"
==> Evaluation "18618691-4d6f-de21-b67c-f7d6d4929319" finished with status "complete"
```

### 2. Check sleep-job status

```
$ nomad status sleep-job
ID          = sleep-job
Name        = sleep-job
Type        = service
Priority    = 50
Datacenters = us-east-1
Status      = <none>

==> Evaluations
ID                                    Priority  TriggeredBy   Status
18618691-4d6f-de21-b67c-f7d6d4929319  50        job-register  complete

==> Allocations
ID                                    EvalID                                NodeID                                TaskGroup    Desired  Status
44640d26-7fc9-6f7d-3d21-2d063102f97d  18618691-4d6f-de21-b67c-f7d6d4929319  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      running
7e43b623-c82b-9f40-765c-0340a3d145a9  18618691-4d6f-de21-b67c-f7d6d4929319  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      running
833cc87a-bbe7-fc59-2ed2-6b3c72237621  18618691-4d6f-de21-b67c-f7d6d4929319  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      running
8ff5c0e5-a9e9-c846-5def-a813ff5eff92  18618691-4d6f-de21-b67c-f7d6d4929319  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      running
fbcdf7df-3d57-fc70-7353-cb1cb9a90540  18618691-4d6f-de21-b67c-f7d6d4929319  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      running
```
### 3. Check sleep-job status again later

After some some time, the sleep finishes, allocations become dead.

```
$ nomad status sleep-job
ID          = sleep-job
Name        = sleep-job
Type        = service
Priority    = 50
Datacenters = us-east-1
Status      = <none>

==> Evaluations
ID                                    Priority  TriggeredBy   Status
18618691-4d6f-de21-b67c-f7d6d4929319  50        job-register  complete

==> Allocations
ID                                    EvalID                                NodeID                                TaskGroup    Desired  Status
44640d26-7fc9-6f7d-3d21-2d063102f97d  18618691-4d6f-de21-b67c-f7d6d4929319  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      dead
7e43b623-c82b-9f40-765c-0340a3d145a9  18618691-4d6f-de21-b67c-f7d6d4929319  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      dead
833cc87a-bbe7-fc59-2ed2-6b3c72237621  18618691-4d6f-de21-b67c-f7d6d4929319  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      dead
8ff5c0e5-a9e9-c846-5def-a813ff5eff92  18618691-4d6f-de21-b67c-f7d6d4929319  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      dead
fbcdf7df-3d57-fc70-7353-cb1cb9a90540  18618691-4d6f-de21-b67c-f7d6d4929319  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      dead
```

### 4. (Optional) Try running sleep-job again

```
$ nomad run sleep.nomad
==> Monitoring evaluation "97abf6be-0735-6ee3-457a-02ea69ca96d9"
    Evaluation triggered by job "sleep-job"
    Allocation "44640d26-7fc9-6f7d-3d21-2d063102f97d" modified: node "f4565321-4370-7a84-06ba-95ec80777c11", group "sleep-group"
    Allocation "7e43b623-c82b-9f40-765c-0340a3d145a9" modified: node "f4565321-4370-7a84-06ba-95ec80777c11", group "sleep-group"
    Allocation "833cc87a-bbe7-fc59-2ed2-6b3c72237621" modified: node "f4565321-4370-7a84-06ba-95ec80777c11", group "sleep-group"
    Allocation "8ff5c0e5-a9e9-c846-5def-a813ff5eff92" modified: node "f4565321-4370-7a84-06ba-95ec80777c11", group "sleep-group"
    Allocation "fbcdf7df-3d57-fc70-7353-cb1cb9a90540" modified: node "f4565321-4370-7a84-06ba-95ec80777c11", group "sleep-group"
    Evaluation status changed: "pending" -> "complete"
==> Evaluation "97abf6be-0735-6ee3-457a-02ea69ca96d9" finished with status "complete"
```

### 5. (Optional) Check sleep-job status again

```
$ nomad status sleep-job
ID          = sleep-job
Name        = sleep-job
Type        = service
Priority    = 50
Datacenters = us-east-1
Status      = <none>

==> Evaluations
ID                                    Priority  TriggeredBy   Status
97abf6be-0735-6ee3-457a-02ea69ca96d9  50        job-register  complete
18618691-4d6f-de21-b67c-f7d6d4929319  50        job-register  complete

==> Allocations
ID                                    EvalID                                NodeID                                TaskGroup    Desired  Status
44640d26-7fc9-6f7d-3d21-2d063102f97d  97abf6be-0735-6ee3-457a-02ea69ca96d9  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      dead
7e43b623-c82b-9f40-765c-0340a3d145a9  97abf6be-0735-6ee3-457a-02ea69ca96d9  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      dead
833cc87a-bbe7-fc59-2ed2-6b3c72237621  97abf6be-0735-6ee3-457a-02ea69ca96d9  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      dead
8ff5c0e5-a9e9-c846-5def-a813ff5eff92  97abf6be-0735-6ee3-457a-02ea69ca96d9  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      dead
fbcdf7df-3d57-fc70-7353-cb1cb9a90540  97abf6be-0735-6ee3-457a-02ea69ca96d9  f4565321-4370-7a84-06ba-95ec80777c11  sleep-group  run      dead
```

Notice that there is a new evaluation that is complete, but no new allocations were made.

### 6. Try restarting nomad

When nomad starts up again, it fails

```
==> Starting Nomad agent...
==> Error starting agent: client setup failed: failed to restore state: 5 error(s) occurred:

* 1 error(s) occurred:

* failed to open ID CGROUP:{"name":"f4fb9752-7f33-36cd-1984-ca1e3ae50305","parent":"","allow_all_devices":true,"allowed_devices":null,"denied_devices":null,"memory":0,"memory_reservation":0,"memory_swap":0,"kernel_memory":0,"cpu_shares":50,"cpuset_cpus":"","cpuset_mems":"","blkio_throttle_read_bps_device":"","blkio_throttle_write_bps_device":"","blkio_throttle_read_iops_device":"","blkio_throttle_write_iops_device":"","blkio_weight":0,"blkio_weight_device":"","freezer":"","hugetlb_limit":null,"slice":"","oom_kill_disable":false,"memory_swappiness":0,"net_prio_ifpriomap":null,"net_cls_classid":""}
: Failed to get pids in the cgroup f4fb9752-7f33-36cd-1984-ca1e3ae50305: open /sys/fs/cgroup/devices/user/1000.user/2.session/f4fb9752-7f33-36cd-1984-ca1e3ae50305/cgroup.procs: no such file or directory
* 1 error(s) occurred:

* failed to open ID CGROUP:{"name":"cedf921e-3628-34be-aa8f-9e84e4feba16","parent":"","allow_all_devices":true,"allowed_devices":null,"denied_devices":null,"memory":0,"memory_reservation":0,"memory_swap":0,"kernel_memory":0,"cpu_shares":50,"cpuset_cpus":"","cpuset_mems":"","blkio_throttle_read_bps_device":"","blkio_throttle_write_bps_device":"","blkio_throttle_read_iops_device":"","blkio_throttle_write_iops_device":"","blkio_weight":0,"blkio_weight_device":"","freezer":"","hugetlb_limit":null,"slice":"","oom_kill_disable":false,"memory_swappiness":0,"net_prio_ifpriomap":null,"net_cls_classid":""}
: Failed to get pids in the cgroup cedf921e-3628-34be-aa8f-9e84e4feba16: open /sys/fs/cgroup/devices/user/1000.user/2.session/cedf921e-3628-34be-aa8f-9e84e4feba16/cgroup.procs: no such file or directory
* 1 error(s) occurred:

* failed to open ID CGROUP:{"name":"590e42ad-ed81-f469-6afc-9117cac499e1","parent":"","allow_all_devices":true,"allowed_devices":null,"denied_devices":null,"memory":0,"memory_reservation":0,"memory_swap":0,"kernel_memory":0,"cpu_shares":50,"cpuset_cpus":"","cpuset_mems":"","blkio_throttle_read_bps_device":"","blkio_throttle_write_bps_device":"","blkio_throttle_read_iops_device":"","blkio_throttle_write_iops_device":"","blkio_weight":0,"blkio_weight_device":"","freezer":"","hugetlb_limit":null,"slice":"","oom_kill_disable":false,"memory_swappiness":0,"net_prio_ifpriomap":null,"net_cls_classid":""}
: Failed to get pids in the cgroup 590e42ad-ed81-f469-6afc-9117cac499e1: open /sys/fs/cgroup/devices/user/1000.user/2.session/590e42ad-ed81-f469-6afc-9117cac499e1/cgroup.procs: no such file or directory
* 1 error(s) occurred:

* failed to open ID CGROUP:{"name":"e8abae82-0730-1a01-8c32-0f6697fc6a3f","parent":"","allow_all_devices":true,"allowed_devices":null,"denied_devices":null,"memory":0,"memory_reservation":0,"memory_swap":0,"kernel_memory":0,"cpu_shares":50,"cpuset_cpus":"","cpuset_mems":"","blkio_throttle_read_bps_device":"","blkio_throttle_write_bps_device":"","blkio_throttle_read_iops_device":"","blkio_throttle_write_iops_device":"","blkio_weight":0,"blkio_weight_device":"","freezer":"","hugetlb_limit":null,"slice":"","oom_kill_disable":false,"memory_swappiness":0,"net_prio_ifpriomap":null,"net_cls_classid":""}
: Failed to get pids in the cgroup e8abae82-0730-1a01-8c32-0f6697fc6a3f: open /sys/fs/cgroup/devices/user/1000.user/2.session/e8abae82-0730-1a01-8c32-0f6697fc6a3f/cgroup.procs: no such file or directory
* 1 error(s) occurred:

* failed to open ID CGROUP:{"name":"db6f2fbb-cee0-8223-ab91-52f99f2a642a","parent":"","allow_all_devices":true,"allowed_devices":null,"denied_devices":null,"memory":0,"memory_reservation":0,"memory_swap":0,"kernel_memory":0,"cpu_shares":50,"cpuset_cpus":"","cpuset_mems":"","blkio_throttle_read_bps_device":"","blkio_throttle_write_bps_device":"","blkio_throttle_read_iops_device":"","blkio_throttle_write_iops_device":"","blkio_weight":0,"blkio_weight_device":"","freezer":"","hugetlb_limit":null,"slice":"","oom_kill_disable":false,"memory_swappiness":0,"net_prio_ifpriomap":null,"net_cls_classid":""}
: Failed to get pids in the cgroup db6f2fbb-cee0-8223-ab91-52f99f2a642a: open /sys/fs/cgroup/devices/user/1000.user/2.session/db6f2fbb-cee0-8223-ab91-52f99f2a642a/cgroup.procs: no such file or directory
```

**Comment by cbednarski** [comment]
Thanks for the report! Unfortunately this is a bug and will be addressed in 0.2. Sorry for the inconvenience!

**Comment by dadgar** [comment]
PR #408
